Assets/LeapMotionModules/Interaction/ActivityManager.cs

```diff
@@ -6,99 +6,118 @@
 
 namespace Leap.Unity.UI.Interaction {
 
-  /// <summary> The Activity Manager encapsulates the logic that determines which
-  /// InteractionBehaviours are within an interactible distance and which interactions
-  /// they should be considered for (hovering/contact/grasping).
-  /// 
-  /// Essentially, the ActivityManager is a wrapper around PhysX sphere queries around
-  /// Hands for InteractionBehaviours.</summary>
-  public class ActivityManager {
-
+  /// <summary>
+  /// ActivityManager is a wrapper around PhysX sphere queries for InteractionBehaviours.
+  /// "Active" objects are objects found in the latest query. It's also possible to get the
+  /// sets of objects that just began or stopped being active since the last query.
+  /// </summary>
+  public class ActivityManager<T> {
+
+    /// <summary> The radius of the query in world-space. </summary>
     public float activationRadius;
-    public InteractionManager manager;
-
-    /// <summary> If non-null, only IInteractionBehaviours for which this function returns true
-    /// will be added to the active behaviours list. </summary>
-    public Func<IInteractionBehaviour, bool> filter = null;
-
-    private Collider[] _colliderResultsBuffer = new Collider[32];
-    private HashSet<IInteractionBehaviour> _activeBehaviours = new HashSet<IInteractionBehaviour>();
 
-    public HashSet<IInteractionBehaviour> ActiveBehaviours {
-      get { return _activeBehaviours; }
-    }
+    /// <summary>
+    /// The layer mask against which to query for active objects. The ActivityManager will only
+    /// find objects in these layers. By default, the ActivityManager will query all layers, but
+    /// this is highly inefficient.
+    /// 
+    /// See SingleLayer and use bitwise operations on their layerMasks for a convenient way to
+    /// express layer masks.
+    /// <see cref="SingleLayer"/>
+    /// </summary>
+    public int activationLayerMask = ~0;
+
+    /// <summary> If non-null, only objects with at least one Collider for which this function
+    /// returns a non-null T will be added to the active behaviours list. </summary>
```

Comment:
Yeah that's not a valid state. And it's because there's no way to know how to convert a Collider reference to a T... I _could_ place a restriction that T has to be a MonoBehaviour and use a default filter of returning Collider.GetComponent<T>? Or even GetComponentInParent<T>? But I'd be worried about default performance in the latter case